digdag-standards/src/main/java/io/digdag/standards/command/ecs/DefaultEcsClient.java

```diff
@@ -322,6 +322,45 @@ public void close()
         throw new RuntimeException("Failed to call EcsClient method");
     }
 
+    /**
+     * Retry function for {@link #getLog(String, String, Optional)}
+     * @param func
+     * @param GetLogEventsRequest
+     * @param <T>
+     * @return T
+     * @throws AmazonServiceException
+     */
+    @VisibleForTesting
+    <T> T retryForGetLog(Supplier<T> func, GetLogEventsRequest request) throws AmazonServiceException
```

Comment:
[minor] How about using `Function<GetLogEventsRequest, T>` instead of `Supplier<T>` so that the caller of this method doesn't need to pass the same `request`s?

```java
    <T> T retryForGetLog(Function<GetLogEventsRequest, T> func, GetLogEventsRequest request) throws AmazonServiceException
    {
        for (int i = 0; i < rateLimitMaxRetry; i++) {
            try {
                return func.apply(request);
            }
          :
    }

        // A caller of the method
        return retryForGetLog((r) -> logs.getLogEvents(r), request);
```

Reply:
fixed https://github.com/treasure-data/digdag/pull/1626/commits/91132878d772f3c45acda9866d678f66a69d2aaa